apps/web/src/components/Basenames/ManageNames/NameDisplay.tsx

```diff
@@ -66,6 +72,17 @@ export default function NameDisplay({
   const openRenewalModal = useCallback(() => setIsRenewalModalOpen(true), []);
   const closeRenewalModal = useCallback(() => setIsRenewalModalOpen(false), []);
 
+  const handleExtendRegistration = useCallback(() => {
+    logEventWithContext('extend_registration_button_clicked', ActionType.click, {
+      context: 'manage_names',
+    });
+    if (isBasenameRenewalsKilled) {
+      openRenewalModal();
+    } else {
+      router.push(`/name/${domain}/renew`);
+    }
```

Comment:
Once renewals launch, the existing modal will be deleted and this logic will be updated to hide the button when the kill switch is on